### DISCUSSION TITLE
As an Admin how to get retention labeled content info for all OneDrive sites in a Tenant using PnP Modern Search?

### DISCUSSION THREAD

**Discussion by Ujjwal06it** [microsoft-search/pnp-modern-search#3530] [Q&A]
**As an Admin how to get retention labeled content info for all OneDrive sites in a Tenant using PnP Modern Search?**
Is it possible using Query Template?

**Comment by wobba** [comment]
See this post https://joannecklein.com/2022/08/25/microsoft-search-add-a-retention-label-filter/ and check out `ComplianceTag`. `ComplianceTag:*` should return all items with a value in that property.

**Reply by Ujjwal06it** [reply]
I would like to get the OneDrive content access to other user's as an admin, how can I get that? your suggested property value only giving the meta data info on my own OneDrive but not other user's OneDrive. @wobba

**Reply by Ujjwal06it** [reply]
I used Search query as {searchTerms} (compliancetag:A* OR compliancetag:B* OR compliancetag:C* OR compliancetag:D* OR compliancetag:E* OR compliancetag:F* OR compliancetag:G* OR compliancetag:H* OR compliancetag:I* OR compliancetag:J* OR compliancetag:K* OR compliancetag:L* OR compliancetag:M* OR compliancetag:N* OR compliancetag:O* OR compliancetag:P* OR compliancetag:Q* OR compliancetag:R* OR compliancetag:S* OR compliancetag:T* OR compliancetag:U* OR compliancetag:V* OR compliancetag:W* OR compliancetag:X* OR compliancetag:Y* OR compliancetag:Z*) CreatedBy={User} SPSiteURL:https://Tenant-my* but this is giving only the result from my own OneDrive. However I want to see result from other user's OneDrive as a SCA. @wobba @stevebeauge @mrjarbenne @PooLP

**Reply by wobba** [reply]
You cannot elevate search via the UX - you have to run something using app-only permissions. Using app-only permissions on search is something you should be very careful about for privacy and compliance reasons.

And an admin does not have access to items via search, which is a good thing. If you really want an admin to see everything you need to add the admin (or group) as an explicit site collection admin and owner on all sites. Then you see everything using your own ACL's.

If I discovered this where I work I would file a law suit.. just saying :) So be careful.
